### DISCUSSION TITLE
U24.02.01-21065

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#872] [Announcements]
Changelog from U23.12.18-20789

- A kot of code refactors and small fixes
- Simplified the aspect ratio scaling logic
- Minor improvements to JitARM64
- Fixed reverb volume in Mario Kart: Double Dash!!
- Minor fixes in the pixel shader and ubershader code
- Updated GameINI fixes

Missing features — these features from Dolphin Official are not yet implemented

- Android input overhaul and subsequent changes to input and TAS handling
- Custom GPU driver support
- Skylanders portal and Infinity base support
- Gamma correction

This codebase (the latest branch) is roughly equivalent to the official Dolphin dev build 5.0-21065 (February 1, 2024). For more information about what has changed check the PR history.

**General information**
- Android developers willing to help are welcome!
- Merry Christmas, everyone! If, by chance, Christmas isn’t a thing where you live, have a great time. Happy gaming!

**Caution**
- **Save states from older versions are, unfortunately, NOT compatible with this release due to significant code refactors. Be sure to make an in-game save before updating!**

**Known Issues**
- The UI in the Cheats section is broken when you try to delete multiple items in a row. This bug is present to this day in Dolphin Official. Just delete a few items, exit the cheats window, and return back.
- For people using touch controls. It is possible that the core controls and touchscreen controls might get out of sync. For more information check here - [https://github.com/Medard22/Dolphin-MMJR2-VBI/discussions/708](https://github.com/Medard22/Dolphin-MMJR2-VBI/discussions/708)

<hr /><em>This discussion was created from the release <a href='https://github.com/Medard22/Dolphin-MMJR2-VBI/releases/tag/U24.02.01-21065'>U24.02.01-21065</a>.</em>

**Comment by Chris353Redfield** [comment]
Hello!
Please make "latching" option for virtual gamepad like in the official version...

**Reply by Medard22** [reply]
The Dolphin version is written in Kotlin on top of the massive input overhaul. I will look into it, but I am still not very familiar with Android development, so no promises.

**Reply by Silverkip1** [reply]
To add another suggestion, please do consider adding an option for sliding buttons like in retroarch (switching from one button to another by sliding your finger,  with an invisible in-between button that presses both at the same time)

**Comment by Lum1r** [comment]
HDMI streaming is not working for me. it crashes app on my device every time. i am not sure if that's this build specific because i just discovered it. Device im using is Anbernic RG Cube.

**Reply by Medard22** [reply]
Interesting... So it crashes when you hook your device to the TV via HDMI right? I can test that on my Retroid. When does it crash? Immediately when you start the app, or after you try to run a game? Are there any other apps or emulators that have similar issues?

**Reply by Lum1r** [reply]
Hi! It crushes if im trying to run any game. I've tried couple displays, few cables and HDMI-USB C converters to exclude hardware issue. It works fine with navigating in menu, browsing options ect. It works just fine on internal display, but once connected to the TV/monitor it immidietaly crashes after launching any game (wii/gamecube) with popup "app stopped working".

**Reply by Medard22** [reply]
Ok, do you know if RG Cube treats its display as portrait or landscape? I remember MMJR2 used to have some weird issues running games in portrait mode. Does it work in the handheld mode without HDMI? Could you turn on logging and try to run a game then post a log here? - (Setting->Log->Write Logs to File, Verbosity - Notice, Enable all log types.) The log would be in root/mmjr2-vbi/Logs/dolphin.log, thanks. I will try it on my Retroid and see if I can reproduce the issue.

**Reply by Medard22** [reply]
I can’t reproduce this issue on my Retroid Pocket 4 Pro. Both the HDMI and USB-C outputs work fine, and I can run and play games without any problems. I’m afraid this might either be device-specific or a bug in the Anbernic firmware. You would need to enable device logging to figure out what is causing the crash. Unfortunately, that’s not something I can do without having the device in hand. Sorry! :(

**Reply by Medard22** [reply]
You could try switching from Vulkan to OpenGL or vice versa. If it´s a GPU driver bug, it might help. You can also check your device settings for anything HDMI related.
